samlang-core/src/interpreter/__tests__/expression-interpreter.test.ts

```diff
@@ -1,775 +1,244 @@
-    classes: methodAccessClasses,
-    localValues: {},
-  };
-  expect(interpreter.eval(methodAccessExpression, methodAccessContext)).toEqual(
-    classMemberFunction
-  );
-  expect(() => interpreter.eval(methodAccessExpression)).toThrow('');
+
+  expect(
+    (new ExpressionInterpreter().eval(methodAccessExpression, {
+      classes: {
+        C: {
+          functions: {},
+          methods: {
+            method: {
+              type: 'functionValue',
+              arguments: [],
+              body: getExpression('5'),
+              context: EMPTY,
+            },
+          },
+        },
+      },
+      localValues: {},
+    }) as FunctionValue).type
+  ).toBe('functionValue');
+  expect(() => new ExpressionInterpreter().eval(methodAccessExpression, EMPTY)).toThrow('');
 });
 
 it('unary expression evaluates correctly', () => {
-  const unaryExpressionNeg = EXPRESSION_UNARY({
-    range: exampleRange,
-    type: intType,
-    operator: '-',
-    expression: intLiteralExpression,
-  });
-  const unaryExpressionNot = EXPRESSION_UNARY({
-    range: exampleRange,
-    type: intType,
-    operator: '!',
-    expression: boolLiteralExpression,
-  });
-  expect(interpreter.eval(unaryExpressionNeg)).toEqual(BigInt(-5));
-  expect(interpreter.eval(unaryExpressionNot)).toEqual(false);
+  expect(interpret('-5')).toEqual(BigInt(-5));
+  expect(interpret('!true')).toEqual(false);
 });
 
 it('panic expression evaluates correctly', () => {
-  const panicExpression = EXPRESSION_PANIC({
-    range: exampleRange,
-    type: stringType,
-    expression: stringLiteralExpression,
-  });
-  expect(() => interpreter.eval(panicExpression)).toThrow('value');
+  expect(() => interpret('panic("value")')).toThrow('value');
 });
 
 it('built in function call expression evaluates correctly', () => {
-  const stringToIntFunctionCall = EXPRESSION_BUILTIN_FUNCTION_CALL({
-    range: exampleRange,
-    type: stringType,
-    functionName: 'stringToInt',
-    argumentExpression: intLiteralExpression,
-  });
-  const stringToIntFunctionCallFail = EXPRESSION_BUILTIN_FUNCTION_CALL({
-    range: exampleRange,
-    type: stringType,
-    functionName: 'stringToInt',
-    argumentExpression: stringLiteralExpression,
-  });
-  const intToStringFunctionCall = EXPRESSION_BUILTIN_FUNCTION_CALL({
-    range: exampleRange,
-    type: intType,
-    functionName: 'intToString',
-    argumentExpression: EXPRESSION_STRING(exampleRange, '5'),
-  });
-  const printlnFunctionCall = EXPRESSION_BUILTIN_FUNCTION_CALL({
-    range: exampleRange,
-    type: stringType,
-    functionName: 'println',
-    argumentExpression: stringLiteralExpression,
+  // FIXME: the test clearly shows that the stringToInt and intToString got messed up.
+  expect(interpret('stringToInt(5)')).toEqual(BigInt(5));
+  expect(() => interpret('stringToInt("value")')).toThrow(`Cannot convert \`value\` to int.`);
+  expect(interpret('intToString("5")')).toEqual('5');
```

Comment:
This does seem to be messed up... when I call `stringToInt` or `intToString`, I didn't actually check that the arguments are indeed `string` or `int` respectively. Is this planning to be fixed in a future PR?

Reply:
Yes in another PR after this one is merged.